File: algorithm/js/select_sort.js
/**
 * 选择排序，每次找出一个最小的元素放到前面
 * @param {Array} arr 待排序数组
 * @returns {Array} arr 排好序的数组
 */
function selectSort(arr) {
  for (let i = 0; i < arr.length; i++) {
    let minIndex = i;
    for (let j = i+1; j < arr.length; j++) {
      if (arr[j] < arr[minIndex]) {
        minIndex = j
      }
    }
    if (i !== minIndex) {
      let temp = arr[i];
      arr[i] = arr[minIndex];
      arr[minIndex] = temp;
    }
  }

  return arr;
}

const arr1 = [1,3,5,8,3,4,0,2,12,43,23,87,54];
console.log(selectSort(arr1));

File: algorithm/go/selectsort/select_sort.go
package main

import (
	"fmt"
)

func selectSort(arr []int) []int {
	length := len(arr)
	for i := 0; i < length; i++ {
		minIndex := i
		for j := i + 1; j < length; j++ {
			if arr[j] < arr[minIndex] {
				minIndex = j
			}
		}

		if i != minIndex {
			temp := arr[i]
			arr[i] = arr[minIndex]
			arr[minIndex] = temp
		}
	}

	return arr
}

func main() {
	arr1 := []int{1, 2, 3, 4, 5, 6, 7, 8}
	arr2 := []int{9, 8, 7, 6, 5, 4, 3, 2, 1}
	arr3 := []int{100, 50, 70, 12, 49, 89, 23, 50, 25}

	fmt.Println(selectSort(arr1))
	fmt.Println(selectSort(arr2))
	fmt.Println(selectSort(arr3))
}